### PR TITLE
Feature/span count sampling

### PR DESCRIPTION
**Pull request by DavidPessoa**
**Description:** Tail sampling strategy was added to sample based on a minimum number of Spans per trace within a batch.  If in a batch all the traces have a number of spans bellow the threshold the batch is discarded.

**Testing:** Tests were added on the config function and sampling strategy. On the sampling strategy, we test if a batch is sampled or not based on the number of spans of all the traces in the batch.

**Documentation:** Readme file was updated to contain new sampling stategy.

### CODE REVIEW

File: CHANGELOG.md
Comment (pmm-sumo) [inline CHANGELOG.md:29]: This should go line 17 or so (it's an enhancement and also one for the unreleased version)

**Comment by pmm-sumo** [comment]
Thank you @DavidPessoa There are some nits and also `gocritic` is failing with some minor items:
```
Error: internal/sampling/span_count_sampler_test.go:76:32: captLocal: `NumberSpans' should not be capitalized (gocritic)
func newTraceWithMultipleSpans(NumberSpans []int32) *TraceData {
                               ^
Error: internal/sampling/span_count_sampler_test.go:81:3: commentFormatting: put a space between `//` and comment text (gocritic)
		//Creates trace
		^
```

Looks good otherwise

**Comment by DavidPessoa** [comment]
@pmm-sumo I've fixed all the issues I believe

File: CHANGELOG.md
Comment (pmm-sumo) [inline CHANGELOG.md:37]: I guess an artifact of merging with the code changes? Should be removed

**Comment by pmm-sumo** [comment]
> @pmm-sumo When are we merging this branch? Otherwise going to have to keep resolving changelog conflicts 😄

I found some more nits after updating with the base branch changes. Typically, when the PR is approved those minor changelog conflicts are fixed by the maintainer. We need one more review too from approver/maintainer

**Comment by DavidPessoa** [comment]
Just did another commit to fix the changelog could you confirm it's correct? Thank you

File: CHANGELOG.md
Comment (pmm-sumo) [reply CHANGELOG.md:49]: @DavidPessoa this part still needs to be removed :)

**Review by pmm-sumo — APPROVED** [review]

**Comment by DavidPessoa** [comment]
@djaglowski Could you review this PR or indicate another person to review it?
Thank you very much

**Pull request merged**
